tests/linalg_test.py

```diff
@@ -560,11 +560,30 @@ def norm(x):
 
     self._CompileAndCheck(partial(jnp.linalg.svd, full_matrices=full_matrices, compute_uv=compute_uv),
                           args_maker)
-    if not (compute_uv and full_matrices):
+    if not compute_uv:
       svd = partial(jnp.linalg.svd, full_matrices=full_matrices,
                     compute_uv=compute_uv)
       # TODO(phawkins): these tolerances seem very loose.
-      jtu.check_jvp(svd, partial(jvp, svd), (a,), rtol=5e-2, atol=2e-1)
+      if dtype == np.complex128:
+        jtu.check_jvp(svd, partial(jvp, svd), (a,), rtol=1e-4, atol=1e-4, eps=1e-8)
+      else:
+        jtu.check_jvp(svd, partial(jvp, svd), (a,), rtol=5e-2, atol=2e-1)
+
+    if compute_uv and (not full_matrices):
+      b, = args_maker()
+      def f(x):
+        u, s, v = jnp.linalg.svd(
+          a + x * b,
+          full_matrices=full_matrices,
+          compute_uv=compute_uv)
+        vdiag = jnp.vectorize(jnp.diag, signature='(k)->(k,k)')
+        return jnp.matmul(jnp.matmul(u, vdiag(s)), v).real
+      _, t_out = jvp(f, (1.,), (1.,))
+      if dtype == np.complex128:
+          tol = 1e-13
+      else:
+          tol = 5e-4
+      self.assertTrue(np.all(np.linalg.norm(t_out - b.real, axis=(-2, -1)) < tol))
```

Comment:
This test is failing on GPU for `complex64` and on TPU for `float32`, which is blocking merge. Because of the use of `assertTrue` the error does not report what the computed tolerance is. It would be better to chnge this to
```python
self.assertLess(np.all(np.linalg.norm(t_out - b.real, axis=(-2, -1)), tol))
```
so the error message reports the norm.

Relevant tolerances will then have to be adjusted in order to pass tests & merge.

Reply:
Actually, that was a silly suggestion. I think this would be the way to do it:
```python
self.assertArraysAllClose(np.linalg.norm(t_out - b.real, axis=(-2, -1)), 0, atol=atol)
```

Reply:
.. or perhaps compare the arrays directly, unless there's good reason to do it via a manual norm:
```python
self.assertArraysAllClose(t_out, b.real, atol=atol)
```

Reply:
I wouldn't say there is any reason to use the norm and I will switch to ```assertArraysAllClose```. However the tests don't fail on the GPU I am using, so I don't really know which tolerance to choose. Hopefully the copybara checks reveal useful insights.

Reply:
Thanks!